### DISCUSSION TITLE
How can I add permission to a canvas app via Github actions ?

### DISCUSSION THREAD

**Discussion by LeonardSchwenk** [microsoft/powerplatform-actions#502] [Q&A] [answered]
I have dev and prd environment in dev I'm the owner of the canvas app.

When I deploy the solution where the canvas app is within the canvas app has no permissions and I had to manually add myself to the app via the PowerApps admin portal.

How can I add a security group or something similar to my solution and to the app.

Thanks in advance

**Comment by rpothin** [comment, marked as answer]
Hi @LeonardSchwenk,

I hope you are doing well.

I am not sure if there is currently a way to achieve this using an out-of-the-box action from this repository.

But it does not mean it is not achievable 😉
You can share canvas app with specific permissions using the [set-adminpowerapproleassignment](https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/set-adminpowerapproleassignment?view=pa-ps-latest) command from the [Microsoft.PowerApps.Administration.PowerShell](https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/?view=pa-ps-latest) PowerShell module.

You can find an example of an implementation in the following links:
- [microsoft/coe-alm-accelerator-templates - share-canvas-app-with-aad-group.yml](https://github.com/microsoft/coe-alm-accelerator-templates/blob/d7c09997b129e4f0811e98a39e6ad2aa1724816a/Pipelines/Templates/share-canvas-app-with-aad-group.yml#L10)
- [microsoft/coe-alm-accelerator-templates - canvas-unpack-pack.ps1](https://github.com/microsoft/coe-alm-accelerator-templates/blob/d7c09997b129e4f0811e98a39e6ad2aa1724816a/PowerShell/canvas-unpack-pack.ps1#L65)

I hope it will help you.
Have a great day.
Raph

**Comment by LeonardSchwenk** [comment]
@rpothin true MVP Raph thanks very helpful info and as polite as a true Canadian !

Cheers,
Leo
